### ISSUE TITLE
[2017.7.3]: git.latest has regression for windows hosts where TypeError exception is now thrown.

### ISSUE THREAD

**Issue by poliva83** [saltstack/salt#46018]
### Description of Issue/Question

https://github.com/saltstack/salt/pull/43769 introduced with 2017.7.3 and has caused regression in the git module for windows hosts. This is because line https://github.com/saltstack/salt/pull/43769/files#diff-59a19c9338fb59c2d90efccd4a0d618fR298 calls `__salt__['file.remove']` passing in `None`. This triggers uncaught TypeError exception to occur here https://github.com/saltstack/salt/blob/v2017.7.3/salt/modules/win_file.py#L1059. You can produce the same error pretty easy on a windows host (see steps to produce).

### Setup

example state file:
```
git_checkout_serverspec_repo:
  git.latest:
    - name: {{ repo }}
    - rev: {{ branch }}
    - branch: {{ branch }}
    - target: {{ settings.serverspec_repo_dir }}
{% if repo.startswith('ssh') %}
    - identity: {{ test_repo_ssh_key_fpath }}
{% elif repo.startswith('https') %}
    - https_user: {{ salt['pillar.get']('salt:serverspec:test_repo:https_user', None) }}
    - https_pass: {{ salt['pillar.get']('salt:serverspec:test_repo:https_pass', None) }}
{% endif %}
    - force_reset: True
    - force_fetch: True
    - force_clone: {{ force_clone }}
```
example state failure:
```
"git_|-git_checkout_serverspec_repo_|-ssh://git@code.td.com/cloud/salt.git_|-latest": {
            "comment": "An exception occurred in this state: Traceback (most recent call last):\n  File \"c:\\salt\\bin\\lib\\site-packages\\salt\\state.py\", line 1851, in call\n    **cdata['kwargs'])\n  File \"c:\\salt\\bin\\lib\\site-packages\\salt\\loader.py\", line 1795, in wrapper\n    return f(*args, **kwargs)\n  File \"c:\\salt\\bin\\lib\\site-packages\\salt\\states\\git.py\", line 701, in latest\n    saltenv=__env__)\n  File \"c:\\salt\\bin\\lib\\site-packages\\salt\\modules\\git.py\", line 3486, in remote_refs\n    saltenv=saltenv)['stdout']\n  File \"c:\\salt\\bin\\lib\\site-packages\\salt\\modules\\git.py\", line 295, in _git_run\n    __salt__['file.remove'](tmp_ssh_wrapper)\n  File \"c:\\salt\\bin\\lib\\site-packages\\salt\\modules\\win_file.py\", line 1059, in remove\n    path = os.path.expanduser(path)\n  File \"c:\\salt\\bin\\lib\\ntpath.py\", line 289, in expanduser\n    if path[:1] != '~':\nTypeError: 'NoneType' object has no attribute '__getitem__'\n", 
            "name": "ssh://git@code.td.com/cloud/salt.git", 
            "start_time": "16:40:54.809000", 
            "result": false, 
            "duration": 4601.0, 
            "__run_num__": 20, 
            "__sls__": "salt.override.test", 
            "changes": {}, 
            "__id__": "git_checkout_serverspec_repo"
        }, 
```

### Steps to Reproduce Issue
1. Open the interactive python on windows host and pass `None` to os.path.expanduser method.
```
PS C:\Users\olivap5> C:\salt\bin\python.exe
Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.path.expanduser(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\salt\bin\lib\ntpath.py", line 289, in expanduser
    if path[:1] != '~':
TypeError: 'NoneType' object has no attribute '__getitem__'
```

### Versions Report

```
PS C:\Users\Administrator> salt-call --versions-report
Salt Version:
Salt: 2017.7.3

Dependency Versions:
cffi: 1.10.0
cherrypy: unknown
dateutil: 2.6.0
docker-py: Not Installed
gitdb: 2.0.3
gitpython: 2.1.3
ioflo: Not Installed
Jinja2: 2.9.6
libgit2: Not Installed
libnacl: 1.6.1
M2Crypto: Not Installed
Mako: 1.0.6
msgpack-pure: Not Installed
msgpack-python: 0.4.8
mysql-python: Not Installed
pycparser: 2.17
pycrypto: 2.6.1
pycryptodome: Not Installed
pygit2: Not Installed
Python: 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)]
python-gnupg: 0.4.0
PyYAML: 3.11
PyZMQ: 16.0.2
RAET: Not Installed
smmap: 2.0.3
timelib: 0.2.4
Tornado: 4.5.1
ZMQ: 4.1.6

System Versions:
dist:
locale: cp1252
machine: AMD64
release: 2012ServerR2
system: Windows
version: 2012ServerR2 6.3.9600 Multiprocessor Free
```

**Comment by poliva83** [comment]
Exact same problem with `__salt__['file.remove'](tmp_identity_file)` call as well: https://github.com/saltstack/salt/pull/43769/files#diff-59a19c9338fb59c2d90efccd4a0d618fR308

**Comment by garethgreenaway** [comment]
@poliva83 Thanks for the report.  The right approach would definitely be to only call `file.remove` if tmp_ssh_wrapper is not None.  We would definitely welcome a PR with this fix.

**Comment by terminalmage** [comment]
Fixed in #46036.
